src/deploy/functions/build.ts

```diff
@@ -224,12 +224,36 @@ export async function resolveBackend(
   const projectId = userEnvOpt.projectId;
   let paramValues: Record<string, Field<string | number | boolean>> = {};
   if (previews.functionsparams) {
-    paramValues = await params.resolveParams(build.params, projectId, userEnvs);
+    paramValues = await params.resolveParams(build.params, projectId, envWithTypes(userEnvs));
+  }
+  if (previews.parampersistence) {
+    const toWrite: Record<string, string> = {};
+    for (const paramName of Object.keys(paramValues).filter((pn) => !userEnvs.hasOwnProperty(pn))) {
```

Comment:
nit* Since we are already have a for loop, I think writing out the filter condition make the code more readable (to me):

```
for (const paramName of Object.keys(paramValues)) {
  if (userEnvs.hasOwnProperty(paramName)) {
     continue;
  }
   toWrite[paramName] = paramValues[paramName]?.toString() || "";
}
```

Reply:
Yeah, I'd agree. Fixed.